Source/DafnyDriver/DafnyDriver.cs

```diff
@@ -302,7 +317,49 @@ public enum CommandLineArgumentsResult {
       }
 
       string programName = dafnyFileNames.Count == 1 ? dafnyFileNames[0] : "the_program";
-      string err = Dafny.Main.ParseCheck(dafnyFiles, programName, reporter, out var dafnyProgram);
+      Program dafnyProgram;
+      string err;
+      if (Action == DriverAction.Format) {
+        if (dafnyFiles.Count == 0) {
+          // Let's list all the dafny files recursively in the working directory
+          dafnyFiles = Directory.GetFiles(Directory.GetCurrentDirectory(), "*.dfy", SearchOption.AllDirectories)
+            .Select(name => new DafnyFile(name)).ToList();
+        }
+
+        var errors = 0;
+        var unchanged = 0;
+        foreach (var dafnyFile in dafnyFiles) {
+          // Might not be totally optimized but let's do that for now
+          err = Dafny.Main.Parse(new List<DafnyFile> { dafnyFile }, programName, reporter, out dafnyProgram);
+          if (err != null) {
+            exitValue = ExitValue.DAFNY_ERROR;
+            Console.Error.WriteLine(err);
+            errors += 1;
+          } else {
+            var firstToken = dafnyProgram.GetFirstTopLevelToken();
+            if (firstToken != null) {
+              var result = Formatting.__default.printSourceReindent(firstToken,
+                IndentationFormatter.ForProgram(dafnyProgram));
+              if (DafnyOptions.O.PrintFile == "-") {
+                Console.Out.Write(result);
+              } else {
+                WriteFile(dafnyFile.FilePath, result);
+              }
+            } else {
+              Console.Error.WriteLine("Could not find text there");
+              errors += 1;
+            }
+          }
+        }
+
+        if (dafnyFiles.Count != 1) {
+          Console.Out.WriteLine("Formatted " + (dafnyFiles.Count - errors) + " files, ignored " + errors + " files");
```

Comment:
Rather than "ignored", we tried and failed to parse them, right? If that's the case then maybe we should say "failed to parse" to be precise.

Reply:
Good idea. I also changed the code to include the files with errors.